azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -600,10 +620,18 @@ func expandNetworkConnectionMonitorEndpoint(input []interface{}) *[]network.Conn
 			result.Address = utils.String(address.(string))
 		}
 
-		if resourceId := v["virtual_machine_id"]; resourceId != "" {
+		if resourceId := v["resource_id"]; resourceId != "" {
 			result.ResourceID = utils.String(resourceId.(string))
 		}
 
+		if endpointType := v["type"]; endpointType != "" {
+			result.Type = network.EndpointType(endpointType.(string))
```

Comment:
As above, what's the default here when `Type` is blank?

Reply:
There is no default value when this property isn't specified.